### PR TITLE
Replace map.filterKeys with map.filter to prevent a memory leak in De…

### PR DESCRIPTION
**Pull request by azolotko**
## Purpose

To fix a memory leak in `DeathWatch.removeFromMap`.

## References

References #27459

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by helena** [comment]
OK TO TEST

File: akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala [hunk 1]
```diff
@@ -116,7 +116,7 @@ private[akka] trait DeathWatch { this: ActorCell =>
   @silent
   private def removeFromMap[T](subject: ActorRef, map: Map[ActorRef, T]): Map[ActorRef, T] =
     if (subject.path.uid != ActorCell.undefinedUid) (map - subject) - new UndefinedUidActorRef(subject)
-    else map.filterKeys(_.path != subject.path).toMap
+    else map.filter { case (k, _) => k.path != subject.path }
```
Comment (helena) [inline akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala]: 👍 Nice. I guess we could have added a minor back port of https://github.com/akka/akka/pull/26831/files#diff-d52fdc08563814c25f721051d7c238c6L114-L119 when it was removed for 2.5. Thanks for contributing this.

**Review by helena — APPROVED** [review]
LGTM

**Review by raboof — APPROVED** [review]
Good one, thanks!

**Pull request merged**